src/test/mockOutputChannel.ts

```diff
@@ -0,0 +1,48 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OutputChannel } from 'vscode'
+
+export class MockOutputChannel implements OutputChannel {
+  public value: string = ''
+  public isHidden: boolean = false
+  public preserveFocus: boolean = false
+
+  public readonly name = 'Mock channel'
+
+  public append(value: string): void {
+    this.value += value
+  }
+
+  public appendLine(value: string) {
+    this.value += value + '\n'
+  }
+
+  public clear(): void {
+    this.value = ''
+  }
+
+  public dispose(): void {
+    this.value = ''
+  }
+
+  public hide(): void {
+    this.isHidden = true
+  }
+
+  /**
+   * This is overloaded by VS Code but viewColumn version is deprecated thus
+   * show(preserveFocus?: boolean) is really what we should consider.
+   * @param args
+   */
+  public show(...args: any[] /* viewColumn?: ViewColumn, preserveFocus?: boolean */) {
```

Comment:
See the 'Overloads' section [here](https://www.typescriptlang.org/docs/handbook/functions.html) for the correct way to implement overloads in TypeScript.

Reply:
Fixed the types by defining them in a manner similar to VS Code.